go-apps/meep-loc-serv/server/loc-serv.go

```diff
@@ -3810,3 +3834,67 @@ func distanceGet(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	fmt.Fprintf(w, string(jsonResponse))
 }
+
+func mec011AppTerminationPost(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	var notification AppTerminationNotification
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&notification)
+	if err != nil {
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if !appEnablementEnabled {
+		//just ignore the message
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+
+	stopRegistrationTicker()
+
+	//delete any registration it made
+	// cannot unsubscribe otherwise, the app-enablement server fails when receiving the confirm_terminate since it believes it never registered
+	//_ = unsubscribeAppTermination(serviceAppInstanceId)
+	_ = deregisterService(serviceAppInstanceId, appEnablementServiceId)
+
+	//send scenario update with a deletion
+	var event scc.Event
+	var eventScenarioUpdate scc.EventScenarioUpdate
+	var process scc.Process
+	var nodeDataUnion scc.NodeDataUnion
+	var node scc.ScenarioNode
+
+	process.Name = instanceName
+	process.Type_ = "EDGE-APP"
+
+	nodeDataUnion.Process = &process
+
+	node.Type_ = "EDGE-APP"
+	node.Parent = mepName
+	node.NodeDataUnion = &nodeDataUnion
+
+	eventScenarioUpdate.Node = &node
+	eventScenarioUpdate.Action = "REMOVE"
+
+	event.EventScenarioUpdate = &eventScenarioUpdate
+	event.Type_ = "SCENARIO-UPDATE"
```

Comment:
We should deregister the service & send the scenario update event within the go routine. This will allow us to return immediately and unblock the App enablement microservice.